core/tracker/tracker_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tracker_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	eth2v1 "github.com/attestantio/go-eth2-client/api/v1"
+	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/core"
+	"github.com/obolnetwork/charon/core/tracker"
+	"github.com/obolnetwork/charon/testutil"
+)
+
+func TestNewTracker(t *testing.T) {
+	duty, defSet := trackerHelper(t)
+
+	deadlineFunc := func(startTime time.Time) func(core.Duty) time.Time {
+		return func(duty core.Duty) time.Time {
+			duration := time.Millisecond
+			lateFactor := 1
+
+			if duty.Type == core.DutyExit {
```

Comment:
you do not need this logic in tests if you do not actually actaully test DutyExit.

Reply:
also since your duration is so small, you can simplify this a lot by just returning time.Time{}, which should result in immediate timeout...?

Reply:
Yeah, but i'm actually passing a `DutyExit` in the test

Reply:
What about reducing it to a single line?
```go
	deadlineFuncProvider := func() func(duty core.Duty) time.Time {
		return func(duty core.Duty) time.Time {
			return time.Now().Add(time.Millisecond * time.Duration(duty.Slot))
		}
	}
```

Reply:
Resolving this conversation as we now use a mock deadliner implementation.